Cabal/src/Distribution/Simple/GHC/Build/ExtraSources.hs

```diff
@@ -86,7 +95,15 @@ buildCSources mbMainFile =
 buildCxxSources mbMainFile =
   buildExtraSources
     "C++ Sources"
-    Internal.componentCxxGhcOptions
+    ( \verbosity lbi bi clbi odir filename ->
+        (Internal.sourcesGhcOptions verbosity lbi bi clbi odir filename)
+          { -- C++ compiler options: GHC >= 8.10 requires -optcxx, older requires -optc
+            -- we want to be able to support cxx-options and cc-options separately
+            ghcOptCcOptions = Internal.separateGhcOptions (mkVersion [8, 10]) (compiler lbi) (Internal.defaultGhcOptCcOptions lbi bi)
+          , -- there are problems with linking with versions lower than 9.4, that's why we need this duplication
+            ghcOptCcProgram = Internal.defaultGhcOptCcProgram lbi
```

Comment:
Same: C-options touched inside a Cxx-function. If this is related to the optcxx vs optc mishmash pre 8.10, I need some sort of simple example to understand how what's done here (and above) helps with that mishmash.

Reply:
I should somehow refer to the tests `cabal-testsuite/PackageTests/FFI/ForeignOptsCxx`